### DISCUSSION TITLE
Force native camera app to use Fossify gallery app

### DISCUSSION THREAD

**Discussion by yaxunil2** [FossifyOrg/General-Discussion#393] [Q&A]
On my phone (Xiaomi Redmi Note 13 5g) after taking a photo and then clicking on the photo thumbnail (within the camera app) the camera opens the 'Native Gallery' instead of the 'Fossify Gallery'? Even though I set the Fossify Gallery as the 'Default Gallery'?? The 'Native Gallery' can not be deleted and if I remove all the permissions and disable it, A notification to restore permissions opens when I click the thumbnail?? Any body know how to resolve that? Thanks

**Comment by DeadSecTr** [comment]
Try disable standart app for gallery in settings (xiaomi app)

**Reply by yaxunil2** [reply]
Already done that - clear cache, remove all permissions, force stop (it's a system app and can not be deleted!) the camera app still wants to open the native gallery app! and asks for permissions, when I cancel it returns to the camera. 
Strange I can access the camera from within the fossify gallery!? but not the other way???

**Comment by naveensingh** [comment]
@yaxunil2 if you are still facing this issue and don't know how to use adb, you can try the following tool to disable the native gallery: https://github.com/0x192/universal-android-debloater

**Reply by yaxunil2** [reply]
Alas it does not help! 
Installed UAD and used it to uninstalled the native gallery app (com.miui.gallery) Re-start the phone. Open camera app - Take a photo, 
When I click the thumbnail no gallery app opens and the camera freeze. Closing the camera app and re-open it un-freezes the app. 
I assume the native gallery app is 'hard coded' in the camera app? Maybe if I could change the name of your app "org.fossify.gallery" to "com.miui.gallery" that would trick the camera app to open your app?? Uhm probably have to also change the name of the native gallery app so that there aren't 2 apps with the same name, as my phone is not rooted that is probably not possible. Any other suggestion or work-around  - thanks 
PS: If it helps the camera app is called: 'com.android.camera' version: 5.1.001120.8

**Reply by naveensingh** [reply]
> I assume the native gallery app is 'hard coded' in the camera app?

Probably yes. Can't think of anything other than rooting the device or switching to a custom ROM. You can try updating if there are any system updates available, maybe they have fixed the camera app.
